### PR TITLE
Remove constructor declaration from PreviewLinkInterface

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Remove constructor declaration from PreviewLinkInterface

#### Why?

Because otherwise this leads to a `Declaration incompatible` error with doctrine proxies

### CODE REVIEW

File: src/Sulu/Bundle/PreviewBundle/Domain/Model/PreviewLink.php [hunk 1]
```diff
@@ -53,15 +53,6 @@ class PreviewLink implements PreviewLinkInterface
      */
     private $lastVisit;
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PreviewBundle/Domain/Model/PreviewLink.php]: Would still keep the constructor this way:

```suggestion

    public function __construct(string $token, string $resourceKey, string $resourceId, string $locale, array $options)
    {
        $this->token = $token;
        $this->resourceKey = $resourceKey;
        $this->resourceId = $resourceId;
        $this->locale = $locale;
        $this->options = $options;
    }

```

File: src/Sulu/Bundle/PreviewBundle/Domain/Model/PreviewLink.php [hunk 2]
```diff
@@ -109,4 +100,20 @@ public function getLastVisit(): ?\DateTimeImmutable
...
+    {
+        $previewLink = new self();
+
+        $previewLink->token = $token;
+        $previewLink->resourceKey = $resourceKey;
+        $previewLink->resourceId = $resourceId;
+        $previewLink->locale = $locale;
+        $previewLink->options = $options;
+
+        return $previewLink;
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PreviewBundle/Domain/Model/PreviewLink.php]:
```suggestion
        return new self(
            $token,
            $resourceKey,
            $resourceId,
            $locale,
            $options
        );
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/PreviewBundle/Infrastructure/Doctrine/Repository/PreviewLinkRepository.php [hunk 3]
```diff
@@ -39,10 +39,10 @@ public function create(string $resourceKey, string $resourceId, string $locale,
     {
         $token = $this->generateToken();
 
-        /** @var class-string<PreviewLinkInterface> $className */
+        /** @var class-string<PreviewLinkInterface>|PreviewLinkInterface $className */
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PreviewBundle/Infrastructure/Doctrine/Repository/PreviewLinkRepository.php]: This is a false indicator as it says its an object and not a string.

I could also not find why it is required when trying to report here something back to phpstan: https://phpstan.org/r/9095f567-476d-4834-951e-a6fe666a7584

And phpstan seems correctly understand class-string of an interface.

Comment (luca-rath) [reply src/Sulu/Bundle/PreviewBundle/Infrastructure/Doctrine/Repository/PreviewLinkRepository.php]:
```suggestion
        /** @var class-string<PreviewLinkInterface>$className */
```

File: src/Sulu/Bundle/PreviewBundle/Infrastructure/Doctrine/Repository/PreviewLinkRepository.php [hunk 4]
```diff
@@ -39,10 +39,10 @@ public function create(string $resourceKey, string $resourceId, string $locale,
     {
         $token = $this->generateToken();
 
-        /** @var class-string<PreviewLinkInterface> $className */
+        /** @var class-string<PreviewLinkInterface>$className */
```
Comment (luca-rath) [inline src/Sulu/Bundle/PreviewBundle/Infrastructure/Doctrine/Repository/PreviewLinkRepository.php]:
```suggestion
        /** @var class-string<PreviewLinkInterface> $className */
```

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**